multisig-transactions-monitor/src/agent.js

```diff
@@ -0,0 +1,128 @@
+const initializeData = {};
+
+function provideInitialize(data) {
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+
+    // get protocol and developer info
+    data.protocolName = config.protocolName;
+    data.protocolAbbreviation = config.protocolAbbreviation;
+    data.developerAbbreviation = config.developerAbbreviation;
+
+    const sigTypeFull = ethers.utils.FormatTypes.full;
+
+    // get contracts' abi and monitored event signatures
+    const { contracts } = config;
+    data.contractsInfo = Object.entries(contracts).map(([name, entry]) => {
+      const { events: eventNames, abiFile } = entry;
+      const file = utils.getAbi(abiFile);
+      const contractInterface = new ethers.utils.Interface(file.abi);
+      const eventSignatures = eventNames.map((eventName) => {
+        const signature = contractInterface.getEvent(eventName).format(sigTypeFull);
+        return signature;
+      });
+
+      // create object to store and return necessary contract information
+      const contract = {
+        name,
+        eventSignatures,
+        eventNames,
+      };
+
+      return contract;
+    });
+
+    // get contrat addresses
+    data.multisigAddress = (config.contracts.multisig.address).toLowerCase();
+    data.governanceAddress = (config.contracts.governance.address).toLowerCase();
+    data.comptrollerAddress = (config.contracts.comptroller.address).toLowerCase();
+  };
+}
+
+function provideHandleTransaction(data) {
+  return async function handleTransaction(txEvent) {
+    const {
+      protocolName,
+      protocolAbbreviation,
+      developerAbbreviation,
+      contractsInfo,
+      multisigAddress,
+      governanceAddress,
+      comptrollerAddress,
+    } = data;
+
+    const findings = [];
+
+    // filter for transactions involving the multisig address
+    const txAddrs = Object.keys(txEvent.addresses).map((address) => address.toLowerCase());
+
+    // if the multisig was involed in a transaction, find out specifically which one
+    if (txAddrs.indexOf(multisigAddress) !== -1) {
+      contractsInfo.forEach((contract) => {
+        // filter for which event and address the multisig transaction was involved in
+        const parsedLogsMultiSig = txEvent.filterLog(contract.eventSignatures, multisigAddress);
+        const parsedLogsGovernance = txEvent.filterLog(contract.eventSignatures, governanceAddress);
+        const parsedLogsComptroller = txEvent.filterLog(
+          contract.eventSignatures,
+          comptrollerAddress,
+        );
+
+        if (parsedLogsMultiSig.length !== 0) {
+          // case if interaction is with gnosis wallet (add/remove owner)
+          parsedLogsMultiSig.forEach((log) => {
+            const finding = utils.createGnosisFinding(
+              log,
+              protocolName,
+              protocolAbbreviation,
+              developerAbbreviation,
+            );
+            findings.push(finding);
+          });
+        }
```

Comment:
There is no need to test if the list contains elements, if it doesn't the iterator will be a NOP 

```suggestion
        // case if interaction is with gnosis wallet (add/remove owner)
        parsedLogsMultiSig.forEach((log) => {
          findings.push(utils.createGnosisFinding(
            log,
            protocolName,
            protocolAbbreviation,
            developerAbbreviation,
          ));
        });
```